java/src/test/android/app/build.gradle

```diff
@@ -31,6 +40,22 @@ android {
 	kotlinOptions {
 		jvmTarget = '1.8'
 	}
+	// Conditional packagingOptions for QNN builds only
+	if (qnnVersion != null)	{
+		packagingOptions  {
+			jniLibs {
+				useLegacyPackaging = true
+			}
+			exclude 'lib/arm64-v8a/libQnnGpu.so'
+		}
+	}
+	sourceSets {
+        main {
+            if (System.properties['qnnVersion'] != null) {
+                manifest.srcFile 'src/main/AndroidManifestQnn.xml'  // Use QNN manifest
+            }
+        }
```

Comment:
nit: there a few lines here indented with spaces and the rest of the file is using tabs.